src/main/java/com/dabsquared/gitlabjenkins/webhook/build/MergeRequestBuildAction.java

```diff
@@ -29,7 +29,12 @@ public MergeRequestBuildAction(Job<?, ?> project, String json) {
         this.mergeRequestHook = JsonUtil.read(json, MergeRequestHook.class);
     }
 
-    public void execute(StaplerResponse response) {
+    /**
+    * Noop for merge request hooks.
+    */
+    void processForCompatibility() {}
```

Comment:
The reason I made `BuildWebHookAction` was because other build hook types could also need to use the compatibility steps.

For example, merge request webhooks have a bunch of fields that only exist in 8.x versions of gitlab and not the 7.x versions, such as source->homepage and source->url  (both of which we end up using in `MergeRequestHookTriggerHandlerImpl`). I added an implementation for this method for MergeRequestBuildAction in d9a66ae5ba77268b4324bddefd1dfdbb545a28c3.